app/javascript/components/ansible-credentials-form/index.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { API } from '../../http_api';
+import MiqFormRenderer from '@@ddf';
+import createSchema from './ansible-credentials-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const AnsibleCredentialsForm = ({ recordId }) => {
+  const [{ fields, initialValues, isLoading }, setState] = useState({ fields: [], isLoading: !!recordId });
+  const promise = useMemo(() => API.options('/api/authentications'), []);
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  const loadSchema = (value, appendState = {}) => ({
+    data: {
+      credential_types: { embedded_ansible_credential_types },
+    },
+  }) =>
+    setState(state => ({
+      ...state,
+      ...appendState,
+      fields: embedded_ansible_credential_types[value].attributes,
+    }));
+
+  useEffect(() => {
+    if (recordId) {
```

Comment:
I was thinking about the missing `manager_resource` attribute and maybe it's better not to touch the API as the endpoint might be used by other resources as well. So you should do an additional request here that's also in the angular controller to retrieve the URI of the provider. It is necessary to bind the two requests together, even though one of them will be conditional and call the `loadSchema` just once.

Something like:
```jsx
API.get('/api/providers?collection_class=ManageIQ::Providers::EmbeddedAutomationManager').then(({ resources: [manager_resource] }) => {
  // Create a dummy promise that resolves to an empty object if the recordId is not set, i.e. creating a new item
  const promise = recordId ? API.get(`/api/authentications/${recordId}`) : Promise.resolve({});
  promise.then(loadSchema(initialValues.type, {
    initialValues: {
      ...initialValues,
      manager_resource,
    },
    isLoading: false,
  })).catch(() => {
    // Some kind of error handling, probably a redirect back with an red error flash message
  });
});
```

Just make sure that after this change the initial value of `isLoading` should be `true` unconditionally as we're going to always load the `manager_resource`.